nixos/modules/services/networking/firewall.nix

```diff
@@ -64,12 +86,18 @@ let
     # Flush the old firewall rules.  !!! Ideally, updating the
     # firewall would be atomic.  Apparently that's possible
     # with iptables-restore.
-    ip46tables -D INPUT -j nixos-fw 2> /dev/null || true
-    for chain in nixos-fw nixos-fw-accept nixos-fw-log-refuse nixos-fw-refuse; do
+    ip46tables -D INPUT -j nixos-fw-input 2> /dev/null || true
+    ip46tables -D OUTPUT -j nixos-fw-output 2> /dev/null || true
+    ip46tables -D FORWARD -j nixos-fw-forward 2> /dev/null || true
+    for chain in nixos-fw nixos-fw-accept nixos-fw-input nixos-fw-output nixos-fw-forward nixos-fw-accept nixos-fw-log-refuse nixos-fw-refuse __invalid_chain__ FW_REFUSE; do
```

Comment:
- `nixos-fw-accept` appears twice
- You can drop `FW_REFUSE` (see 460b43dbfe8f21253637be350eb1ef1d586eaf5e (was after your PR))
- I don't really like the following name: `__invalid_chain__` (that's just my personal opinion tho)